typescript/web/src/components/import-button/import-images-modal/__tests__/import-images-modal.tsx

```diff
@@ -1,96 +1,63 @@
 /* eslint-disable import/first */
 import { act, render, screen, waitFor } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
-import { ApolloProvider } from "@apollo/client";
+import { MockedProvider } from "@apollo/client/testing";
 import { PropsWithChildren } from "react";
 import "@testing-library/jest-dom/extend-expect";
-import { processImage } from "../../../../connectors/repository/image-processing";
-import { client } from "../../../../connectors/apollo-client/schema-client";
-import { setupTestsWithLocalDatabase } from "../../../../utils/setup-local-db-tests";
 import {
   mockUseQueryParams,
   mockNextRouter,
 } from "../../../../utils/router-mocks";
+import { BASIC_DATASET_DATA } from "../../../../utils/tests/data.fixtures";
 
 mockUseQueryParams();
 mockNextRouter({
-  query: { datasetSlug: "test-dataset", workspaceSlug: "local" },
+  isReady: true,
+  query: {
+    datasetSlug: BASIC_DATASET_DATA.slug,
+    workspaceSlug: BASIC_DATASET_DATA.workspace.slug,
+  },
 });
 
 import { ImportImagesModal } from "../import-images-modal";
-import { CREATE_LOCAL_TEST_DATASET_MUTATION } from "../../../../utils/tests/mutations";
+import { getApolloMockLink } from "../../../../utils/tests/apollo-mock";
+import { ERROR_MOCKS, IMPORT_BUTTON_MOCKS } from "../../import-button.fixtures";
 
 const files = [
   new File(["Hello"], "hello.png", { type: "image/png" }),
   new File(["World"], "world.png", { type: "image/png" }),
   new File(["Error"], "error.pdf", { type: "application/pdf" }),
 ];
 
-const Wrapper = ({ children }: PropsWithChildren<{}>) => (
-  <ApolloProvider client={client}>{children}</ApolloProvider>
+type WrapperProps = PropsWithChildren<{
+  error?: boolean;
+}>;
+const Wrapper = ({ error = false, children }: WrapperProps) => (
+  <MockedProvider
+    link={getApolloMockLink(error ? ERROR_MOCKS : IMPORT_BUTTON_MOCKS)}
+  >
+    {children}
+  </MockedProvider>
 );
```

Comment:
```suggestion
const createWrapper =
  (error: boolean = false) =>
  ({ children }: PropsWithChildren<{}>) => (
    <MockedProvider
      link={getApolloMockLink(error ? ERROR_MOCKS : IMPORT_BUTTON_MOCKS)}
    >
      {children}
    </MockedProvider>
  );
```

Reply:
Why would it be better to use this form ?

Reply:
It keeps the wrapper being a React component with only a single `children` prop and prevents calling it like a function as you did after (hence the other code suggestion).